modules/core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/CoreService.java

```diff
@@ -75,75 +75,57 @@ public void deactivate(ComponentContext context) {
     }
 
     @Override
-    public void registerContribution(Object contrib, String point, ComponentInstance contributor) {
-        if (VERSION_REMOVAL_POLICY_XP.equals(point)) {
-            registerVersionRemovalPolicy((CoreServicePolicyDescriptor) contrib);
-        } else if (ORPHAN_VERSION_REMOVAL_FILTER_XP.equals(point)) {
-            registerOrphanVersionRemovalFilter((CoreServiceOrphanVersionRemovalFilterDescriptor) contrib);
-        } else {
-            throw new RuntimeException("Unknown extension point: " + point);
-        }
+    public void start(ComponentContext context) {
+        versionRemovalPolicy = this.<CoreServicePolicyDescriptor> getRegistryContribution(VERSION_REMOVAL_POLICY_XP)
+                                   .map(this::createVersionRemovalPolicy)
+                                   .orElse(null);
+        orphanVersionRemovalFilters.addAll(
+                this.<CoreServiceOrphanVersionRemovalFilterDescriptor> getRegistryContributions(
+                        ORPHAN_VERSION_REMOVAL_FILTER_XP)
+                    .stream()
+                    .map(this::createOrphanVersionRemovalFilter)
+                    .collect(Collectors.toList()));
```

Comment:
What do you think about assigning the result to `orphanVersionRemovalFilters` and nullifying it in the `stop` method?

Reply:
I've been using that pattern to avoid having API calls that would be done before component start to end with a NPE, but maybe i should assume no call should be done until the component is started anyway?

Reply:
Done changes in separate commits to ease up re-review (will fixup commits after validation)